cmd/gcp-controller-manager/node_annotator.go

```diff
@@ -188,9 +191,20 @@ func (na *nodeAnnotator) processNextWorkItem() bool {
 	}
 	defer na.queue.Done(key)
 
-	na.sync(key.(string))
+	err := na.sync(key.(string))
+	if err != nil {
+		// Drop from the queue if failing too many times.
+		if na.queue.NumRequeues(key) > nodeAnnotatorQueueRetryLimit {
+			klog.Errorf("Sync %v exceeded retry limit, last err: %v", key, err)
+			na.queue.Forget(key)
+			return true
+		}
```

Comment:
The queue already has a per-key rate limit with backoff. We should not give up, just slow down. Controller logic should be level based:

https://github.com/kubernetes/community/blob/master/contributors/design-proposals/architecture/principles.md#control-logic

Reply:
Seems retrying forever fits with reconcilation model better.
@danielywong I'm also going to drop the retrylimit in node_syncer